src/codeflare_sdk/utils/generate_yaml.py

```diff
@@ -191,6 +202,82 @@ def update_nodes(
                 update_resources(spec, min_cpu, max_cpu, min_memory, max_memory, gpu)
 
 
+def update_ca_secret(ca_secret_item, cluster_name, namespace):
+    from . import generate_cert
+
+    metadata = ca_secret_item.get("generictemplate", {}).get("metadata")
+    metadata["name"] = f"ca-secret-{cluster_name}"
+    metadata["namespace"] = namespace
+    metadata["labels"]["odh-ray-cluster-service"] = f"{cluster_name}-head-svc"
+    data = ca_secret_item.get("generictemplate", {}).get("data")
+    data["ca.key"], data["ca.crt"] = generate_cert.generate_ca_cert(365)
+
+
+def enable_local_interactive(resources, cluster_name, namespace):
+    rayclient_route_item = resources["resources"].get("GenericItems")[2]
+    ca_secret_item = resources["resources"].get("GenericItems")[3]
+    item = resources["resources"].get("GenericItems")[0]
+    update_rayclient_route(rayclient_route_item, cluster_name, namespace)
+    update_ca_secret(ca_secret_item, cluster_name, namespace)
+    # update_ca_secret_volumes
+    item["generictemplate"]["spec"]["headGroupSpec"]["template"]["spec"]["volumes"][0][
+        "secret"
+    ]["secretName"] = f"ca-secret-{cluster_name}"
+    item["generictemplate"]["spec"]["workerGroupSpecs"][0]["template"]["spec"][
+        "volumes"
+    ][0]["secret"]["secretName"] = f"ca-secret-{cluster_name}"
+    # update_tls_env
+    item["generictemplate"]["spec"]["headGroupSpec"]["template"]["spec"]["containers"][
+        0
+    ]["env"][1]["value"] = "1"
+    item["generictemplate"]["spec"]["workerGroupSpecs"][0]["template"]["spec"][
+        "containers"
+    ][0]["env"][1]["value"] = "1"
+    # update_init_container
+    command = item["generictemplate"]["spec"]["headGroupSpec"]["template"]["spec"][
+        "initContainers"
+    ][0].get("command")[2]
+
+    command = command.replace("deployment-name", cluster_name)
+
+    server_name = (
+        oc.whoami("--show-server").split(":")[1].split("//")[1].replace("api", "apps")
```

Comment:
Will eventually have to be changed to kubernetes client, can be done at same time as `cluster.local_client_url()` though, no need in this PR (maybe add a note in the new issue #173)

Reply:
noted.